### PR TITLE
feat(*) kuma-dp verifies connection to the kuma-cp

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

Since Bootstrap is now served via HTTPS, we should verify their cert.

I introduced a new arg to `kuma-dp` called `ca-cert-file`. When it's specified and the connection to bootstrap is HTTPS (now by default) then we verify cert.
If bootstrap uses HTTPS but we don't specify this file, we've got a warning in logs:
```
Connecting to HTTPS Bootstrap server without CA verification. Consider passing CA via --ca-cert-file or KUMA_CONTROL_PLANE_CA_CERT.
```

On Kubernetes, injector now automatically pass CA to every dataplane. I introduced new config for HELM (and kumactl install control plane) that let's you specify CA cert that then will be used for verification.

Keep in mind that DP first connects to API Server for catalog that returns path to Boostrap server. Right now is API Server is HTTP without TLS.
That means that technically someone can impersonate Control Plane and provide invalid bootstrap path, but the attacker would have to acquire cert for invalid bootstrap server signed by CA.
Overall I think we should connect directly to the bootstrap server which should have the same port for XDS, SDS, but it's not in the scope of this PR.

### Documentation

- [ ] TODO (production deployment guide?)

### CODE REVIEW

File: docs/cmd/kumactl/HELP.md [hunk 1]
```diff
@@ -246,6 +246,7 @@ Flags:
       --kds-global-address string           URL of Global Kuma CP (example: grpcs://192.168.0.1:5685)
       --mode string                         kuma cp modes: one of standalone|remote|global (default "standalone")
       --namespace string                    namespace to install Kuma Control Plane to (default "kuma-system")
+      --tls-ca-cert string                  CA certificate that was used to generate TLS certificate for Kuma Control Plane servers
```
Comment (austince) [inline docs/cmd/kumactl/HELP.md:249]: Can you add this to `deployments/charts/kuma/README.md` as well?

**Review by austince — COMMENTED** [review]
Looks good to me in terms of the chart!

File: pkg/config/plugins/runtime/k8s/config.go [hunk 2]
```diff
@@ -97,8 +97,10 @@ type Injector struct {
 	VirtualProbesEnabled bool `yaml:"virtualProbesEnabled" envconfig:"kuma_runtime_kubernetes_virtual_probes_enabled"`
 	// VirtualProbesPort is a port for exposing virtual probes which are not secured by mTLS
 	VirtualProbesPort uint32 `yaml:"virtualProbesPort" envconfig:"kuma_runtime_kubernetes_virtual_probes_enabled"`
-	// Cnfiguration for a traffic that is intercepted by sidecar
+	// Configuration for a traffic that is intercepted by sidecar
```
Comment (lobkovilya) [inline pkg/config/plugins/runtime/k8s/config.go:100]: I know that you just fixed a typo, but let's make it according to the guide for comments in Go - `// SidecarTraffic is a configuration ...`

File: pkg/plugins/runtime/k8s/webhooks/injector/injector.go [hunk 3]
```diff
@@ -212,6 +223,10 @@ func (i *KumaInjector) NewSidecarContainer(pod *kube_core.Pod, ns *kube_core.Nam
 				Name:  "KUMA_DATAPLANE_RUNTIME_TOKEN_PATH",
 				Value: "/var/run/secrets/kubernetes.io/serviceaccount/token",
 			},
+			{
+				Name:  "KUMA_CONTROL_PLANE_CA_CERT",
+				Value: i.caCert,
```
Comment (lobkovilya) [inline pkg/plugins/runtime/k8s/webhooks/injector/injector.go:228]: Is it secure that Kuma CP injects CA cert which will be used to verify Kuma CP? Can someone put their own CA to `CaCertFile` and impersonate `bootstrap` at the end?
Can we use Kubernetes Secrets and mount them into the sidecar?

Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go:228]: Secrets are Namespaced Scope, we would have to populate this secret all over the namespaces.
CA cert is something public so I see no problem with placing this as an env var.

If someone has a control over Control Plane, it's done. They can do whatever they can.

Comment (lobkovilya) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go:228]: You don't need control over Control Plane, Control Plane reads CA from `CaCertFile` I think it's possible to put malicious CA here. 
Yes, CA cert is something public, but I think you should trust the source who provides you this CA. But in this case, we validate Kuma CP with CA provided by Kuma CP.

Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go:228]: You need to change CaCertFile AND Bootstrap URL that we are providing for the attack, therefore you need control over the control plane. Or am I missing some cases here?

Comment (lobkovilya) [reply pkg/plugins/runtime/k8s/webhooks/injector/injector.go:228]: Okay, probably you are right. It just feels weird that the one who you are trying to attest provides you a way how to attest to him. But then I realized it is not Kuma CP provides CA, webhook provides this CA. Injector technically is part of Kubernetes and is trusted by Kubernetes based on certs

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**